jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -351,12 +351,22 @@ private static void arraycopy(Object[] A1, int offset1, Object[] A2, int offset2
 private static final int FileEncoding = 2;
 private static final int OSEncoding = 3;
 
+/*[IF Java11]*/
+// JCL native
+private static native Properties initProperties(Properties props);
+/*[ENDIF] Java11 */
+
 /**
  * If systemProperties is unset, then create a new one based on the values 
  * provided by the virtual machine.
  */
 @SuppressWarnings("nls")
 private static void ensureProperties() {
+/*[IF Java11]*/
+	// invoke JCL native to initialize platform encoding
+	initProperties(new Properties());
```

Comment:
Doesn't the next line setting `systemProperties = new Properties();` throw away the ones that have been initialized?

Reply:
Also looking at the PR for the JDK 11 extensions repo to understand how this fits together.

Reply:
The purpose of this invoking of JCL native is to initialize `platform encoding` only and the `Properties` set is `OpenJDK` specific and can't be used by `OpenJ9`.